src/mtype.h

```diff
@@ -208,12 +216,13 @@ struct mtype {
         std::string death_drops;
         float luminance;           // 0 is default, >0 gives luminance to lightmap
         int hp;
-        std::vector<unsigned int> sp_freq;     // How long sp_attack takes to charge
+        // special attack frequencies and function pointers
+        std::unordered_map<std::string, mtype_special_attack> special_attacks;
+        std::vector<std::string> special_attacks_names; // names of attacks, in json load order
```

Comment:
To be fair, the old system had just as many redundant vector indices as this has redundant attack names.

I discussed the options on IRC with @kevingranade and we couldn't come up with a better way that was reliable. He also OKed some options that would be cleaner here, but would result in [more] occasional incorrect behavior during loading of saves.

Making it private seems obvious in retrospect. I'm not sure if it should only be used when loading old saves, though. In the past we had an undocumented dependency on the order of special attacks in monsters.json. If someone wants to use that ordering (and it's an array, not an object, so the order is a legitimate thing to want to use), this is for them.